x/msgfees/types/proposal.go

```diff
@@ -0,0 +1,145 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	// ProposalTypeAddMsgFee to add a new msg based fee
+	ProposalTypeAddMsgFee string = "AddMsgFee"
+	// ProposalTypeUpdateMsgFee to update an existing msg based fee
+	ProposalTypeUpdateMsgFee string = "UpdateMsgFee"
+	// ProposalTypeRemoveMsgFee to remove an existing msg based fee
+	ProposalTypeRemoveMsgFee string = "RemoveMsgFee"
+)
+
+var (
+	_ govtypes.Content = &AddMsgFeeProposal{}
+	_ govtypes.Content = &UpdateMsgFeeProposal{}
+	_ govtypes.Content = &RemoveMsgFeeProposal{}
+)
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeAddMsgFee)
+	govtypes.RegisterProposalTypeCodec(AddMsgFeeProposal{}, "provenance/msgfees/AddMsgFeeProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeUpdateMsgFee)
+	govtypes.RegisterProposalTypeCodec(UpdateMsgFeeProposal{}, "provenance/msgfees/UpdateMsgFeeProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeRemoveMsgFee)
+	govtypes.RegisterProposalTypeCodec(RemoveMsgFeeProposal{}, "provenance/msgfees/RemoveMsgFeeProposal")
+}
+
+func NewAddMsgFeeProposal(
+	title string,
+	description string,
+	msg string,
+	additionalFee sdk.Coin) *AddMsgFeeProposal {
+	return &AddMsgFeeProposal{
+		Title:         title,
+		Description:   description,
+		MsgTypeUrl:    msg,
+		AdditionalFee: additionalFee,
+	}
+}
+
+func (ambfp AddMsgFeeProposal) ProposalRoute() string { return RouterKey }
+func (ambfp AddMsgFeeProposal) ProposalType() string  { return ProposalTypeAddMsgFee }
+func (ambfp AddMsgFeeProposal) ValidateBasic() error {
+	if len(ambfp.MsgTypeUrl) == 0 {
+		return ErrEmptyMsgType
+	}
+
+	if !ambfp.AdditionalFee.IsPositive() {
+		return ErrInvalidFee
+	}
+
+	return govtypes.ValidateAbstract(&ambfp)
+}
+func (ambfp AddMsgFeeProposal) String() string {
```

Comment:
Probably need to run `make lint`.

Reply:
only results i get from `make lint` are 
```
x/metadata/types/address.go:178:10: appendAssign: append result not assigned to the same slice (gocritic)
	addr := append(SessionKeyPrefix, bz...)
	        ^
x/metadata/types/address.go:192:10: appendAssign: append result not assigned to the same slice (gocritic)
	addr := append(RecordKeyPrefix, bz...)
	        ^
x/metadata/types/address.go:225:10: appendAssign: append result not assigned to the same slice (gocritic)
	addr := append(RecordSpecificationKeyPrefix, bz...)
	        ^
x/name/legacy/v042/store.go:35:17: appendAssign: append result not assigned to the same slice (gocritic)
		updatedKey := append(updateAddress, nameKey...)
		              ^
x/attribute/types/keys.go:34:9: appendAssign: append result not assigned to the same slice (gocritic)
	key := append(AttributeKeyPrefix, address.MustLengthPrefix(acc.Bytes())...)
	       ^
x/attribute/types/keys.go:46:9: appendAssign: append result not assigned to the same slice (gocritic)
	key := append(AttributeKeyPrefix, address.MustLengthPrefix(acc.Bytes())...)
	       ^
x/name/keeper/keeper.go:102:14: appendAssign: append result not assigned to the same slice (gocritic)
	indexKey := append(addrPrefix, key...) // [0x04] :: [addr-bytes] :: [name-key-bytes]
	            ^
x/name/keeper/keeper.go:199:14: appendAssign: append result not assigned to the same slice (gocritic)
	indexKey := append(addrPrefix, key...) // [0x02] :: [addr-bytes] :: [name-key-bytes]
	            ^
x/attribute/legacy/v042/types.go:21:9: appendAssign: append result not assigned to the same slice (gocritic)
	key := append(AttributeKeyPrefixLegacy, acc.Bytes()...)
	       ^
make: *** [lint] Error 1
```

Reply:
The `make lint` command does 3 things.
1. Runs `golang-ci`.
2. Runs `gofmt -s -d` on the go files.
3. Runs `go mod verify`.

Those are from `golang-ci`, and they prevent the other 2 things from running. I'm guessing you've got a version of `golang-ci` other than `v1.39`, which is why you're getting these errors, but the github action isn't.

The `gofmt` step should be complaining about not having an empty line here.

A run of `make lint` shouldn't identify anything. It should just look like this:
```
$ make lint
/usr/local/bin/golangci-lint run
find . -name '*.go' -type f -not -path "./vendor*" -not -path "./client/*" -not -path "*.git*" -not -path "*.pb.go" | xargs gofmt -d -s
/usr/local/bin/go mod verify
all modules verified
```